cli/install/install.go

```diff
@@ -0,0 +1,800 @@
+		log.Infof("The operation requires some dependencies.")
+		fmt.Println("Missing packages: " + strings.Join(missing_pack, " ") + " " +
+			strings.Join(missing_pack_src, " "))
+		if osName == "darwin" {
+			fmt.Println("You can install them by running commands:")
+			fmt.Println("brew install " + strings.Join(missing_pack, " ") +
+				strings.Join(missing_pack_src, " "))
+		} else if strings.Contains(osName, "CentOs") {
+			fmt.Println("You can install them by running command:")
+			if len(missing_pack) != 0 {
+				fmt.Println(" sudo yum install " + strings.Join(missing_pack, " "))
+			}
+			if len(missing_pack_src) != 0 {
+				fmt.Println("install from sources: " +
+					strings.Join(missing_pack_src, " "))
+			}
+		} else if strings.Contains(osName, "Ubuntu") || strings.Contains(osName, "Debian") {
+			fmt.Println("You can install them by running command:")
+			if len(missing_pack) != 0 {
+				fmt.Println(" sudo apt install " + strings.Join(missing_pack, " "))
+			}
+			if len(missing_pack_src) != 0 {
+				fmt.Println("install from sources: " +
+					strings.Join(missing_pack_src, " "))
+			}
+		}
+		fmt.Println("Usage: tt install -f if you already have those packages installed")
+		return false
+	}
+	return true
+}
+
+// checkExisting checks if program is already installed in binary directory.
+func checkExisting(version string, dst string) bool {
+	if _, err := os.Stat(filepath.Join(dst, version)); os.IsNotExist(err) {
+		return false
+	} else {
+		return true
+	}
+}
+
+// ExecuteCommand executes program with given args in verbose or quiet mode.
+func ExecuteCommand(program string, isVerbose bool, logFile *os.File, workDir string,
+	args ...string) error {
+	cmd := exec.Command(program, args...)
+	if isVerbose {
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+	} else {
+		cmd.Stdout = logFile
+		cmd.Stderr = logFile
+	}
+	if workDir == "" {
+		workDir, _ = os.Getwd()
+	}
+	cmd.Dir = workDir
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	err = cmd.Wait()
+	return err
+}
+
+// downloadRepo downloads git repository.
+func downloadRepo(repoLink string, tag string, dst string,
+	logFile *os.File, verbose bool) error {
+	var err error
+	if tag == "master" {
+		err = util.ExecuteCommand("git", verbose, logFile, dst, "clone",
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	} else {
+		err = util.ExecuteCommand("git", verbose, logFile, dst, "clone", "-b", tag,
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	}
+
+	return err
+}
+
+// copyBuildedTT copies tt binary.
+func copyBuildedTT(dst, path, version string, flags InstallationFlag,
+	logFile *os.File) error {
+	var err error
+	if _, err := os.Stat(dst); os.IsNotExist(err) {
+		err = os.Mkdir(dst, FolderRights)
+		if err != nil {
+			return fmt.Errorf("Unable to create bindir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		err = os.Remove(filepath.Join(dst, version))
+		if err != nil {
+			printLog(logFile.Name())
```

Comment:
I don't think os.Remove writes something useful to our log. Why do we print it here?

Reply:
removed